engine/execution/ingestion/engine.go

```diff
@@ -558,6 +558,23 @@ func (e *Engine) executeBlock(ctx context.Context, executableBlock *entity.Execu
 		return
 	}
 
+	// if the receipt is for a sealed block, then no need to broadcast it.
+	lastSealed, err := e.state.Sealed().Head()
```

Comment:
does this consider finalization for the block that includes the seal?

Reply:
yes, sealed means it's sealed AND finalized.

[We update `Sealed` in `Finalize` method of the protocol state](https://github.com/onflow/flow-go/blob/d6f84cc9106530060008e1f9853dd00902a10f7a/state/protocol/badger/mutator.go#L718)